test/e2e/common/setup-kubestellar.sh

```diff
@@ -41,10 +41,53 @@ source "$COMMON_SRCS/setup-shell.sh"
 
 :
 : -------------------------------------------------------------------------
-: Create a Kind hosting cluster with nginx ingress controller and KubeFlex controller-manager
+: Create a hosting cluster with nginx ingress controller and KubeFlex controller-manager
 :
-kflex init --create-kind $disable_chatty_status
-: Kubeflex kind cluster created.
+if [ -z "$USE_K3D" ]; then
```

Comment:
@eaepstein With such a big change + we may support other platforms in the future, I would recommend we separate the platform dependent part into separate scripts and have a different setup for each (setup-kind setup-ocp setup-k3d, ect..) 
We of course  should try to leave all common parts in a single script

Reply:
There is really only one big change: creating the kubeflex cluster. All that code should be folded into kubeflex so that 
`kflex init --create-kind $disable_chatty_status`
becomes
`kflex init --create-k3d $disable_chatty_status`

All the other places are one or two line changes which could be externalized into one or more extensible scripts.

Reply:
So maybe we should just start with that ?  and add a PR to KF and only then to here ?